src/redux/groups/groupsSlice.ts

```diff
@@ -1,38 +1,62 @@
+import { TreeNodeData } from '@mantine/core';
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
-import { Groups } from '@/types/types';
+import { Group } from '@/types/types';
+
+export type Groups = {
+  [key: string]: Group;
+};
+
+export type CustomGroupOrdering = {
+  [key: string]: string[]; // group paths
+};
 
 export interface GroupsState {
-  customGroupOrdering: object; // TODO specify this
+  customGroupOrdering: CustomGroupOrdering;
   groups: Groups;
+  sceneTreeData: TreeNodeData[];
+  tags: string[];
+  expandedGroups: string[];
 }
 
 const initialState: GroupsState = {
   customGroupOrdering: {},
-  groups: {}
+  groups: {}, // @TODO (emmbr, 2024-12-04): this could maybe be removed from the state
+  sceneTreeData: [],
+  tags: [],
+  expandedGroups: []
 };
 
 export const groupsSlice = createSlice({
   name: 'groups',
   initialState,
   reducers: {
-    updateCustomGroupOrdering: (
-      state,
-      action: PayloadAction<object> // TODO: make a type?
-    ) => {
+    updateCustomGroupOrdering: (state, action: PayloadAction<CustomGroupOrdering>) => {
       state.customGroupOrdering = action.payload;
-      // Alternatively:
-      // return {
-      //   ...state,
-      //   customGroupOrdering: action.payload
-      // }
+      return state;
     },
     setGroups: (state, action: PayloadAction<Groups>) => {
       state.groups = action.payload;
+    },
+    setSceneTreeData: (state, action: PayloadAction<TreeNodeData[]>) => {
+      state.sceneTreeData = action.payload;
+    },
+    setTags: (state, action: PayloadAction<string[]>) => {
+      state.tags = action.payload;
+    },
+    storeSceneTreeNodeExpanded: (state, action: PayloadAction<string[]>) => {
+      state.expandedGroups = action.payload;
+      return state;
```

Comment:
Good catch on the inconsistencies here. I'll add the `return state;` to the other functions, to be more consistent with the other parts of the code. 

But yeah, we should make a decision, since now we mix between the `return state;` and `return { ... }` version